### PR TITLE
new PR working for ruxit and ggas, adding secondary energy fuels and heat disaggreagation

### PR DESCRIPTION
**Pull request by leggler**
New PR after @sebwehrle and @danielhuppmann comments that H2 can not be a primary fuel

changes:

- added tag_secondary_energy.yaml: fuel carriers very similar to final energy carriers
- added tag_heat.yaml with heat-sources - consistent with approach in tag_electricity.yaml (and related to Issue#7)
- slightly changed and re-arranged tag_primary_energy.yaml and tag_electricity.yaml to be consistent with gas|methane|fossil, solid|fossil, solid|biomass etc. approach chosen in final energy
- added units to tag_final_energy.yaml

### CODE REVIEW

File: definitions/variable/tag_secondary_energy.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        description: Synthesized methane from non-renewable sources
+        unit: TJ
+    - Gas|Biogas:
+        description: Gas from biogenic sources including landfill gas, sewer gas, and 
+          biogas (lower methane content than fossil natural gas)
+        unit: TJ
+    - Gas|Coal Gas:
+        description: Coal gases
+        unit: TJ
+    - Electricity|{Electricity Input}:
```
Comment (danielhuppmann) [inline definitions/variable/tag_secondary_energy.yaml:63]: Recursive tag-lists are not supported by the **nomenclature** package.

File: definitions/variable/energy.yaml [hunk 2]
```diff
@@ -1,10 +1,33 @@
+- Domestic Production|{Primary Energy Carrier}:
```
Comment (danielhuppmann) [inline definitions/variable/energy.yaml:1]: It's quite difficult to review a PR if it's mixing actual changes with simple re-ordering - better to keep those as separate PRs.

File: definitions/variable/energy.yaml [hunk 3]
```diff
@@ -1,10 +1,33 @@
...
+    unit: TJ
+    note: should be equivalent to 'Imports' - 'Exports'
 - Primary Energy|{Primary Energy Carrier}:
     description: Primary energy consumption of {Primary Energy Carrier}
     unit: TJ
     note: should be equal to 'Domestic Production' + 'Imports' - 'Exports'
-- Secondary Energy|Electricity|{Electricity Input}:
-    description: Total net generation of electricity from {Electricity Input}
-      (including transmission losses)
+- Secondary Energy|{Secondary Energy Carrier}:
```
Comment (danielhuppmann) [inline definitions/variable/energy.yaml:29]: Given that the inputs to different secondary-energy-carriers are quite different, it's usually easier to keep those separate, i.e., 
```yaml
- Secondary Energy|Electricity|{Electricity Input}
- Secondary Energy|Hydrogen|{Hydrogen Input}
```

File: definitions/variable/energy.yaml [hunk 4]
```diff
@@ -1,10 +1,33 @@
...
+    description: Energy commodity produced from a primary energy carrier
```
Comment (danielhuppmann) [inline definitions/variable/energy.yaml:30]: It's not true that secondary-energy can only be generated from primary fuels - hydrogen can be generated from electricity...

File: definitions/variable/tag_electricity.yaml [hunk 5]
```diff
@@ -1,5 +1,9 @@
 - Electricity Input:
-    - Coal:
+    - Solids:
```
Comment (danielhuppmann) [inline definitions/variable/tag_electricity.yaml:2]: I would keep that structure as it is for primary and the electricity sector, to avoid overly long variable names...

File: definitions/variable/tag_final_energy.yaml [hunk 6]
```diff
@@ -1,51 +1,74 @@
 - Final Energy Carrier:
     - Solids:
         description: Solid fuels
+        unit: TJ
```
Comment (danielhuppmann) [inline definitions/variable/tag_final_energy.yaml:4]: It's not necessary to add units in the tag-lists unless the variable where that tag-list is used specifically applies it as 
```yaml
- Primary Energy|{Primary Fuels}:
    unit: {Primary Fuels}
```

File: definitions/variable/tag_heat.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    - Gas|Methane|Synthetic|Renewable Sources:
+        description: Synthesized methane from renewable sources
+    - Gas|Methane|Synthetic|Other Sources:
+        description: Synthesized methane from non-renewable sources
+    - Gas|Biogas:
+        description: Gas from biogenic sources including landfill gas, sewer gas, and 
+          biogas (lower methane content than fossil natural gas)
+    - Gas|Coal Gas:
+        description: Coal gases
+    - Hydro:
```
Comment (danielhuppmann) [inline definitions/variable/tag_heat.yaml:41]: Hydro cannot be used for heat generation...

**Review by danielhuppmann — COMMENTED** [review]
I will close this PR because there are several issues, and I will open a dedicated PR with the relevant changes. I added a few comments to better explain how the nomenclature package (and this GitHub repository) can be used more effectively.

**Comment by danielhuppmann** [comment]
Closing this PR, will create one or several dedicated PRs with the relevant changes.

File: definitions/variable/energy.yaml (hunk 2 above)
Comment (leggler) [reply definitions/variable/energy.yaml:1]: still new to the workflow, but i know what you mean... if there only wouldnt be this urge to do everything at once when you finally have time to do it... ;D

File: definitions/variable/energy.yaml (hunk 3 above)
Comment (leggler) [reply definitions/variable/energy.yaml:29]: OK... but shouldnt it be 
`Secondary Energy|Gas|Hydrogen|{Hydrogen Input}`
because thats the same way we split gaseous fuels up in final consumption?

and it becomes a bit more tricky for synthetic methane...

also we would need

`Secondary Energy|Heat|{Heat Input}`

File: definitions/variable/energy.yaml (hunk 4 above)
Comment (leggler) [reply definitions/variable/energy.yaml:30]: You are right - looking it up in the IEA Energy Statistics Manual its defined like that:

> Energy commodities are either extracted or captured directly from natural resources (and are termed primary) (...) or are produced from primary commodities. All energy commodities which are not primary but produced from primary commodities are termed secondary commodities. Secondary energy comes from the transformation of primary **_or_** secondary energy. (...) **Both electricity and heat may be produced in a primary or secondary form.** 

So i would suggest `Energy carrier produced from other energy carriers`

File: definitions/variable/tag_electricity.yaml (hunk 5 above)
Comment (leggler) [reply definitions/variable/tag_electricity.yaml:2]: hrm... the general issue here that 

- some commodities are used across **_all_** flows (e.g. natural gas or solid biofuels: exists as primary, secondary _and_ end use fuel) 
- while others exist **_only in one flow_**, primary (e.g. Anthracite, Crude Oil..) or secondary (e.g. coke oven coke,  manufactured gases, hydrogen...)

and I am uncertain how to deal with it in the terminology. Wouldnt it be confusing if the same thing (ch4) is called `Final Energy|Gas|Methane|Fossil` in final energy, but `Primary|Natural gas` in primary, because some layers of hirarchy are missing? Same goes for `Final Energy|Solid|biomass` and `Primary Energy|Biomass`

Its also difficult for me to judge if this inconsistent hirarchy levels might create any downstream problems when modelling.

ps: In general I dont like the term "Natural Gas", because it implies a postiv framing ("nature"). I prefer german "Erdgas"

File: definitions/variable/tag_heat.yaml (hunk 7 above)
Comment (leggler) [reply definitions/variable/tag_heat.yaml:41]: true, thought I delted them...

File: definitions/variable/tag_electricity.yaml (hunk 5 above)
Comment (danielhuppmann) [reply definitions/variable/tag_electricity.yaml:2]: The structure is trying to strike a balance between readability and clarity. At the primary-energy level, we are usually interested in the "usual" categories (oil, fossil gas, coal, ...) whereas at the final-energy level, we are usually interested in the state of the energy carrier (liquid, gaseous, electricity, hot water) - and it's not really important at the consumer-level how the electricity was generated or how the water was heated. Hence the switch...

For the modelling side, it should not make any difference because all IAMC-variables are just labels and it is the responsibility of the modeler to implement the mapping from the labels to the model structure.

File: definitions/variable/energy.yaml (hunk 3 above)
Comment (danielhuppmann) [reply definitions/variable/energy.yaml:29]: It‘s relatively obvious that at the final-energy-level, hydrogen is a type of gas, but it’s not really necessary to repeat that information for describing the sources from which hydrogen was produced. Hence in the interest of concise variables, it can be omitted.

Also yes, we‘ll add other types of secondary-energy-carriers…
